### ISSUE TITLE
Invalid BOM is generated for certain packages

### ISSUE THREAD

**Issue by maglo** [CycloneDX/cyclonedx-webpack-plugin#27]
### Brief

When building a BOM, for some packages the BOM that is generated contains invalid values, causing `cyclonedx-cli` to fail to parse it.

### Steps to reproduce

Add a reference to `pkg:npm/history@5.1.0` in your webpack project and configure webpack to build a BOM as described in the readme.

Run `cyclonedx-cli analyze --input-file .well-known/sbom --input-format json`

### Expected results

🆗 

### Actual results

```
...
Unhandled exception: System.Text.Json.JsonException: The JSON value could not be converted to CycloneDX.Models.v1_2.Dependency. Path: $.dependencies[19] | LineNumber: 2576 | BytePositionInLine: 12.
...
```

upon inspecting, the dependencies node for `pkg:npm/history@5.1.0` reads

```
#22 0.301       "ref": "pkg:npm/history@5.1.0",
#22 0.301       "dependsOn": [
#22 0.301         null
#22 0.301       ]
#22 0.301     },
```

### Workaround

`sed -i 's/null//' ./.well-known/sbom`

**Comment by jkowalleck** [comment]
according to [CDX JSON spec](https://github.com/CycloneDX/specification/blob/master/schema/bom-1.3.schema.json) `dependsOn`'s items must be of type `string`

@stevespringett is the webpack-plugin adding a `null` as a dependency?

**Comment by maglo** [comment]
Maybe this is interesting:

```
/src/frontend # npm ls history
frontend@0.0.0 /src/frontend
`-- react-router-dom@6.0.2
  +-- history@5.1.0
  `-- react-router@6.0.2
    `-- history@5.1.0  deduped
```

**Comment by stevespringett** [comment]
@maglo I'm having trouble reproducing the issue. Do you have a sample project that exhibits this behavior?

**Comment by maglo** [comment]
Sure!

`webpack.config.js` https://pastebin.com/MSg9Lbk6
 
`package.json` https://pastebin.com/FNsA1j7N

`index.js` https://pastebin.com/VrJFp3ju

`App.js` https://pastebin.com/5ntp8hyK

I run this in a Dockerfile:

```
# Webpack temporary container
FROM node:14.18.2-alpine3.14 as webpack
COPY ./src/frontend /src/frontend
RUN mkdir -p /src/backend/bvs_web/static

# Set up limited environment
RUN adduser -D builder && \
    chown builder:builder -R /src

WORKDIR /src/frontend
USER builder
RUN npm install
RUN npm run build
```

which yields:

```
/src/frontend # grep null -A 5 -B 5 /src/backend/bvs_web/static/frontend/.well-known/sbom
      "ref": "pkg:npm/react-router@6.0.2"
    },
    {
      "ref": "pkg:npm/history@5.1.0",
      "dependsOn": [
        null
      ]
    },
    {}
  ]
}
```

and

```
/src/frontend # cyclonedx-cli analyze --input-file /src/backend/bvs_web/static/frontend/.well-known/sbom --input-format json
Unhandled exception: System.Text.Json.JsonException: The JSON value could not be converted to CycloneDX.Models.v1_2.Dependency. Path: $.dependencies[7] | LineNumber: 352 | BytePositionInLine: 12.
   at CycloneDX.Json.Converters.v1_2.DependencyConverter.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.IEnumerableDefaultConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonConverter jsonConverter, Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at CycloneDX.Json.Deserializer.DeserializeAsync_v1_2(Stream jsonStream)
   at CycloneDX.Json.Deserializer.DeserializeAsync(Stream jsonStream)
   at CycloneDX.Cli.CliUtils.InputBomHelper(String filename, BomFormat format)
   at CycloneDX.Cli.Commands.AnalyzeCommand.Analyze(AnalyzeCommandOptions options)
   at System.CommandLine.Invocation.CommandHandler.GetExitCodeAsync(Object value, InvocationContext context)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass27_0.<<UseVersionOption>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__24_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseDebugDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

**Comment by stevespringett** [comment]
Thank you. I was able to reproduce. Looking into a fix now.

**Comment by stevespringett** [comment]
@maglo I found a few issues that led to nulls being added to dependencies, and an undefined component being added to the inventory. I've resolved these issues and have published v1.0.1 of the webpack plugin. Go ahead and test when you get a chance.

**Comment by maglo** [comment]
works like a charm. issue can be closed if you ask me.

🙇
